lazy-pages/src/sys.rs

```diff
@@ -134,7 +145,14 @@ pub unsafe fn user_signal_handler(info: ExceptionInfo) -> Result<(), Error> {
 
             let page_buf = PageBuf::new_from_vec(buffer_as_slice.to_vec())
                 .expect("Cannot panic here, because we create slice with PageBuf size");
-            page.release(page_buf)?;
+
+            if ctx
+                .released_lazy_pages
+                .insert(page, Some(page_buf))
+                .is_some()
+            {
+                return Err(Error::DoubleRelease(page));
+            }
```

Comment:
the check may be preformed before we do a lots of calculations: as minimum, we can check if released lazy pages contain data under the key before we allocate vec at line 146

Reply:
I cannot see any reasons to do this, because this errors cannot appear in normal cases, only if we have bugs in node or runtime. So, in case they will appear we must fix them immediately

Reply:
Yeah, but even if this is, for example, impossible moving code 5 lines upper is not a big deal in order to optimise it, or just beautify the code

Reply:
Because this part of code uses `page_buf ` we cannot move it up. So we must write separate useless code before.